kotlin/src/main/kotlin/org/axonframework/extensions/kotlin/serialization/AxonSerializers.kt

```diff
@@ -49,8 +49,34 @@ import org.axonframework.messaging.responsetypes.OptionalResponseType
 import org.axonframework.messaging.responsetypes.ResponseType
 import kotlin.reflect.KClass
 
-private val trackingTokenSerializer = PolymorphicSerializer(TrackingToken::class).nullable
+/**
+ * Serializer for Axon's [TrackingToken] class.
+ * Provides serialization and deserialization support for nullable instances of TrackingToken.
+ * This serializer uses [replyTokenContextSerializer] to serialize the context field and now only [String] type or null value is supported!
+ *
+ * @see TrackingToken
+ */
+val trackingTokenSerializer = PolymorphicSerializer(TrackingToken::class).nullable
 
+/**
+ * Serializer for the [ReplayToken.context], represented as a nullable String.
+ * This context is typically used to provide additional information during token replay operations.
+ *
+ * This serializer is used by [trackingTokenSerializer] to serialize the context field and now only [String] type or null value is supported!
+ * Sadly enough, there's no straightforward solution to support [Any]; not without adjusting the context field of the ReplayToken in Axon Framework itself.
+ * That is, however, a breaking change, and as such, cannot be done till version 5.0.0 of the Axon Framework.
+ * This also allow more complex objects as the context, although it requires the user to do the de-/serialization to/from String, instead of the Axon Framework itself.
+ * Look at AxonSerializersTest, case `replay token with complex object as String context` for an example how to handle that using Kotlin Serialization.
+ *
+ * @see ReplayToken.context
+ */
+val replyTokenContextSerializer = String.serializer().nullable
```

Comment:
Isn't there a way in kotlin serialization, as we have full control over (de)serialization, to encode the type of the payload as the first field, and then serialize the object properties next?

Something like outlined here https://stackoverflow.com/questions/66148137/how-to-serialize-any-type-in-kotlinx-serialization

Giving up the flexibility of the type in here feels wrong. But if there's no other way, so be it.

Reply:
In my conversations with @lion7 it seemed impractical to do with the current set up.

We discussed that it would've been best if the `ReplayToken` didn't contain a serializable `Object` at all.
But for example instead the `SerializedObject` from AF.

Adjusting the type **is** the breaking change I was referring too. We can't do that right now at all.